lib/util/ts_parser.js

```diff
@@ -255,26 +257,52 @@ shaka.util.TsParser = class {
         case 0x06:
           // stream_type 6 can mean a lot of different things in case of DVB.
           // We need to look at the descriptors. Right now, we're only
-          // interested in AC-3 and EC-3 audio, so we do the descriptor parsing
-          // only when we don't have an audio PID yet.
-          if (result.audio == -1 && esInfoLength > 0) {
+          // interested in a few audio and video formats,.
+          if (esInfoLength > 0) {
             let parsePos = offset + 5;
             let remaining = esInfoLength;
+            // Descriptor info: https://www.streamguru.de/mpeg-analyzer/supported-descriptor-list/
             while (remaining > 2) {
               const descriptorId = data[parsePos];
+              const descriptorLen = data[parsePos + 1] + 2;
               switch (descriptorId) {
+                // Registration descriptor
+                case 0x05: {
+                  const registrationData =
+                      data.subarray(parsePos + 2, parsePos + descriptorLen);
+                  const registration =
+                      StringUtils.fromCharCode(registrationData);
+                  if (result.audio == -1 && registration === 'Opus') {
+                    result.audio = pid;
+                    result.audioCodec = 'opus';
+                  } else if (result.video == -1 && registration === 'AV01') {
+                    result.audio = pid;
```

Comment:
Shouldn't this be the video and video codec, rather than the audio and audio codec?

Reply:
Fixed!